sdk/python/feast/infra/materialization/snowflake_engine.py

```diff
@@ -274,7 +274,11 @@ def _materialize_one(
 
             fv_latest_values_sql = offline_job.to_sql()
 
-            if feature_view.entity_columns:
+            if (
+                feature_view.entity_columns[0].name == DUMMY_ENTITY_ID
+            ):  # entityless Feature View's placeholder entity
+                entities_to_write = 1
+            else:
```

Comment:
Entityless feature views do have an entity column, it's just that it's a dummy value.